### ISSUE TITLE
A couple of problems running GRIDSS-2.12.0 and gridss_annotate_vcf_kraken2

### ISSUE THREAD

**Issue by toddajohnson** [PapenfussLab/gridss#500]
I just ran  GRIDSS-2.12.0 and ran into a couple of issues.

1) I tried running GRIDSS-2.12.0 'all' steps against some samples that had previously been aligned against GRCh37.fa using bwa mem 0.7.8.  They made it through ExtractSVReads fine, but all aborted at the PreprocessForBreakendAssembly stage with 
`INFO	2021-06-03 18:56:36	StreamingSplitReadRealigner	Seen many non-increasing record positions. Printing Read-names as well.
[E::bns_fetch_seq] begin=3144246914, mid=3144246915, end=3137454505, len=6792409, seq=0x2b0274603010, rid=85, far_beg=3101976562, far_end=3137454505
java: bntseq.c:444: bns_fetch_seq: Assertion `seq && *end - *beg == len' failed.
[W::bgzf_read_block] EOF marker is absent. The input is probably truncated
samtools sort: truncated file. Aborting`

I then went back to the original fastq files, ran alignment to GRCh38 with bwa 0.7.17, and reran gridss, and all finished successfully.  Maybe, that is due to the old bwa version? If so, that it might be good to mention that version dependency on the README page.

2) Following the successful GRIDSS 2.12.0 'all' steps run, I then ran gridss_annotate_vcf_kraken2 and gridss_annotate_vcf_repeatmasker to annotate the output GRIDSS VCFs.   During gridss_annotate_vcf_kraken2, the run core dumped with the logs ending as below.

`[Sat Jun 05 23:54:04 JST 2021] InsertedSequencesToFasta --INPUT /home/tjohnson/workspace/runs/VHL_WGS_HMF_20210530/result/GRIDSS-2.12.0/VHL_01/gridss/VHL_01.gridss.driver.vcf.gz --OUTPUT /dev/stdout --MIN_SEQUENCE_LENGTH 20 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --GA4GH_CLIENT_SECRETS client_secrets.json --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false
[Sat Jun 05 23:54:04 JST 2021] AnnotateVariantsKraken --INPUT /home/tjohnson/workspace/runs/VHL_WGS_HMF_20210530/result/GRIDSS-2.12.0/VHL_01/gridss/VHL_01.gridss.driver.vcf.gz --OUTPUT /home/tjohnson/workspace/runs/VHL_WGS_HMF_20210530/result/GRIDSS-2.12.0/VHL_01/gridss/VHL_01.gridss.kraken2.vcf.gz --KRAKEN_INPUT /dev/stdin --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --GA4GH_CLIENT_SECRETS client_secrets.json --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false
[Sat Jun 05 23:54:04 JST 2021] Executing as tjohnson@gc181i on Linux 3.10.0-1127.18.2.el7.x86_64 amd64; Java HotSpot(TM) 64-Bit Server VM 1.8.0_181-b13; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:2.12.0-gridss
[Sat Jun 05 23:54:04 JST 2021] Executing as tjohnson@gc181i on Linux 3.10.0-1127.18.2.el7.x86_64 amd64; Java HotSpot(TM) 64-Bit Server VM 1.8.0_181-b13; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:2.12.0-gridss
"java -Xmx1g $jvm_args -cp $gridss_jar gridss.kraken.AnnotateVariantsKraken -INPUT "$1" -OUTPUT "$output" -KRAKEN_INPUT /dev/stdin" command completed with exit code 0`

I compared that log with a previous successful GRIDSS-2.11.1 run, and noticed the use of temp files instead of /dev/stdin and /dev/stdout. Replacing lines 167-180 of gridss_annotate_vcf_kraken (where output between gridss.InsertedSequencesToFasta, kraken2, and gridss.kraken.AnnotateVariantsKraken is now piped) with lines 170-178 of gridss_annotate_vcf_kraken2.sh resulted in all runs completing successfully.

**Comment by d-cameron** [comment]
Any chance you should share (privately if necessary) the VCF that's causing the core dump?

>[E::bns_fetch_seq] begin=3144246914, mid=3144246915, end=3137454505, len=6792409, seq=0x2b0274603010, rid=85, far_beg=3101976562, far_end=3137454505 java: bntseq.c:444: bns_fetch_seq: Assertion seq && *end - *beg == len'`

I'm not entirely sure why bwa seg faults but I've hit it myself occasionally and it's very frustrating. Good point about the version checking - I should be more aggressive about erroring out when using a known bad version of bwa.

**Comment by toddajohnson** [comment]
Hi Daniel,
As far point #2 above, I was having a similar problem with VIRUSBreakend, and finally realized that at one point I took the need for assigning JVM heap memory a little too strongly and coded "export _JAVA_OPTIONS="-Xmx${jvmheap}" into my bash script.  I took that out and now VIRUSBreakend is not crashing. With kraken2 taking up over 50G, that extra unneeded forcing of -Xmx probably pushed the memory needs over the 64GB job request when several commands are piped together and caused the server to force quit the job.
